src/arrayMethodSort.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i; j < this.length; j++) {
+        let condition;
+
+        if (compareFunction === undefined) {
+          condition = this[i].toString() > this[j].toString();
+        } else {
+          condition = compareFunction(this[i], this[j]) > 0;
+        }
```

Comment:
it's better to set a default value for the compare function as it will make a code shorter and cleaner 
`[].__proto__.sort2 = function(compareFunction = (a, b) => String(a) > String(b)) {`

Reply:
thanck you